bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java

```diff
@@ -139,6 +140,19 @@ public void receivedMessage(ThingUID connectionBridge, MqttBrokerConnection conn
             return;
         }
 
+        BaseChannelConfiguration config = BaseChannelConfiguration
+                .fromString(new String(payload, StandardCharsets.UTF_8), gson);
```

Comment:
Would you ever receive two json config payloads that have their json fields rearranged but are otherwise equal? If not then you could use the hashCode of the payload directly using `Arrays.hashCode(payload)` instead of using the hashcode of the config. 
By hashing the payload directly you avoid unnecessarily decoding the payload and creating the config object in the case that you exit from the method early.
In addition you wouldn't need to implement the hashCode method in BaseChannelConfiguration.

WDYT?

Reply:
Ummmmm... you are totally right, but first I would like to hear what @jochen314 (or @J-N-K) thinks about my proposed solution. Do you see any drawback functional wise? Usually HA devices should not send the json fields in different order and, also, this topics are only sent once and as retained, so hashing a payload could be enough.

Reply:
@jochen314's opinion is also more than welcomed :wink:

Reply:
First of all, we would need to go with `Arrays.hashCode(payload)` or we would need to add a `hashCode` function to all subsclasses of `BaseChannelConfiguration`. 
And I don't like implementing `hashCode` without implementing `equals`....

Reply:
I would change the implementation to make use of Arrays.hashCode(payload), sure :+1: